operator/src/test/java/org/keycloak/operator/testsuite/unit/KeycloakStatusTest.java

```diff
@@ -17,21 +17,137 @@
+        assertCondition(rolling, null, null, null);
+    }
+
+    @Test
+    public void testErrorWithPriorStatus() {
+        // with prior status, ready and rolling are preserved
+        KeycloakStatus prior = new KeycloakStatusAggregator(1L).build();
+        prior.getConditions().stream().forEach(c -> c.setLastTransitionTime("prior"));
+
+        KeycloakStatus status = new KeycloakStatusAggregator(prior, 2L).addErrorMessage("this is bad").build();
+        KeycloakStatusCondition ready = status.findCondition(KeycloakStatusCondition.READY).orElseThrow();
+        assertCondition(ready, true, 1L, "");
+        assertEquals("prior", ready.getLastTransitionTime());
+
+        KeycloakStatusCondition errors = status.findCondition(KeycloakStatusCondition.HAS_ERRORS).orElseThrow();
+        assertCondition(errors, true, 2L, "this is bad");
+        assertNotEquals("prior", errors.getLastTransitionTime());
+
+        KeycloakStatusCondition rolling = status.findCondition(KeycloakStatusCondition.ROLLING_UPDATE).orElseThrow();
+        assertCondition(rolling, false, 1L, "");
+    }
+
+    @Test
+    public void testReadyWithPriorStatus() {
+        // without prior status, ready and rolling are known and keep the transition time
+        KeycloakStatus prior = new KeycloakStatusAggregator(1L).build();
+        prior.getConditions().stream().forEach(c -> c.setLastTransitionTime("prior"));
+
+        KeycloakStatus status = new KeycloakStatusAggregator(prior, 2L).build();
+        KeycloakStatusCondition ready = status.findCondition(KeycloakStatusCondition.READY).orElseThrow();
+        assertCondition(ready, true, 2L, "");
+        assertEquals("prior", ready.getLastTransitionTime());
+
+        KeycloakStatusCondition errors = status.findCondition(KeycloakStatusCondition.HAS_ERRORS).orElseThrow();
+        assertCondition(errors, false, 2L, "");
+
+        KeycloakStatusCondition rolling = status.findCondition(KeycloakStatusCondition.ROLLING_UPDATE).orElseThrow();
+        assertCondition(rolling, false, 2L, "");
+    }
+
+    @Test
+    public void testPreservesScale() {
+        KeycloakStatus prior = new KeycloakStatusAggregator(1L).apply(b -> b.withObservedGeneration(1L).withInstances(3)).build();
+        prior.getConditions().stream().forEach(c -> c.setLastTransitionTime("prior"));
+
+        KeycloakStatus status = new KeycloakStatusAggregator(prior, 2L).apply(b -> b.withObservedGeneration(2L)).build();
+        assertEquals(2, status.getObservedGeneration());
+        assertEquals(3, status.getInstances());
+    }
+
+    private void assertCondition(KeycloakStatusCondition condition, Boolean status, Long observedGeneration, String message) {
+        assertEquals(status, condition.getStatus());
+        assertEquals(observedGeneration, condition.getObservedGeneration());
+        assertEquals(message, condition.getMessage());
+        if (status != null) {
+            assertNotNull(condition.getLastTransitionTime());
+        }
+    }
```

Comment:
Maybe we could extend/reuse the existing asserts?
https://github.com/keycloak/keycloak/blob/19ee00ff545d0c6cb68079849b5f18188f38928c/operator/src/test/java/org/keycloak/operator/testsuite/utils/CRAssert.java#L33-L56

Reply:
I'll check on how to combine the assertions.

Reply:
The logic has been aligned.